File: board-service/internal/domain/workspace_join_request.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type JoinRequestStatus string

const (
	JoinRequestPending  JoinRequestStatus = "PENDING"
	JoinRequestApproved JoinRequestStatus = "APPROVED"
	JoinRequestRejected JoinRequestStatus = "REJECTED"
)

type WorkspaceJoinRequest struct {
	BaseModel
	WorkspaceID uuid.UUID         `gorm:"type:uuid;not null" json:"workspace_id"`
	UserID      uuid.UUID         `gorm:"type:uuid;not null" json:"user_id"`
	Status      JoinRequestStatus `gorm:"type:varchar(20);not null;default:'PENDING'" json:"status"`
	RequestedAt time.Time         `gorm:"not null;default:CURRENT_TIMESTAMP" json:"requested_at"`
	ProcessedAt *time.Time        `json:"processed_at"`
	ProcessedBy *uuid.UUID        `gorm:"type:uuid" json:"processed_by"`
}

func (WorkspaceJoinRequest) TableName() string {
	return "workspace_join_requests"
}

File: user-service/src/main/java/OrangeCloud/UserRepo/entity/WorkspaceJoinRequest.java
package OrangeCloud.UserRepo.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "workspaceJoinRequests")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString
@EqualsAndHashCode(of = "joinRequestId")
public class WorkspaceJoinRequest {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "joinRequestId", updatable = false, nullable = false, columnDefinition = "UUID")
    private UUID joinRequestId;

    @Column(name = "workspaceId", nullable = false, columnDefinition = "UUID")
    private UUID workspaceId;

    @Column(name = "userId", nullable = false, columnDefinition = "UUID")
    private UUID userId;

    @Column(name = "status", nullable = false)
    @Enumerated(EnumType.STRING)
    @Builder.Default
    private JoinRequestStatus status = JoinRequestStatus.PENDING;

    @CreationTimestamp
    @Column(name = "requestedAt", updatable = false)
    private LocalDateTime requestedAt;

    @UpdateTimestamp
    @Column(name = "updatedAt")
    private LocalDateTime updatedAt;

    public enum JoinRequestStatus {
        PENDING,
        APPROVED,
        REJECTED
    }
}